core/iwasm/include/wasm_export.h

```diff
@@ -1407,6 +1407,70 @@ WASM_RUNTIME_API_EXTERN bool
 wasm_runtime_unregister_natives(const char *module_name,
                                 NativeSymbol *native_symbols);
 
+struct WASMGlobalInstance;
+typedef struct WASMGlobalInstance *wasm_global_instance_t;
+/**
+ * Get an export global
+ *
+ * @param exec_env the execution environment to retrieve
+ * @param module_name the export module name
+ * @param name the export global name
+ *
+ * @return the export global
+ *
+ */
+wasm_global_instance_t
+wasm_runtime_get_export_global_instance(wasm_exec_env_t exec_env,
+                                        const char *name);
+
+/**
+ * Get the kind of a global instance
+ *
+ * @param export_global the global instance
+ *
+ * @return the kind of the global instance
+ *
+ */
+wasm_valkind_t
+wasm_runtime_global_instance_get_kind(wasm_global_instance_t export_global);
+
+/**
+ * Get the mutability of a global instance
+ *
+ * @param export_global the global instance
+ *
+ * @return true if mutable, false otherwise
+ *
+ */
+bool
+wasm_runtime_global_instance_get_mutable(wasm_global_instance_t export_global);
+
+/**
+ * Get the value of a global instance
+ *
+ * @param export_global the global instance
+ * @param value the pre-allocated pointer to store the value
+ *
+ * @return true if successful, false otherwise
+ *
+ */
+bool
+wasm_runtime_global_instance_get_value(wasm_global_instance_t export_global,
+                                       wasm_val_t *value);
+
+/**
+ * Set the value of a global instance
+ *
+ * @param export_global the global instance
+ * @param value pointer to the value to store
+ *
+ * @return true if successful, false otherwise
+ *
+ */
+bool
+wasm_runtime_global_instance_set_value(wasm_global_instance_t export_global,
+                                       const wasm_val_t *value);
+
```

Comment:
reading is fine. I am not sure there should be a writable API.

Reply:
@lum1n0us can you please explain that a little more? I would think that in the case where the global is declared as mutable and exported from the WASM file, the intention is for it to be writable from outside of WASM code.

Reply:
Yes. It's fine if a wasm module is designed to be host-wasm co-operated.

Reply:
The returned global_inst already has `void *global_data` field, developer can read/write from/to it, seems no need to add extra write API?